cdc/log.cc

```diff
@@ -845,21 +863,27 @@ class transformer final {
         std::transform(pc.begin(), pc.end(), std::back_inserter(columns), [](auto& c) { return &c; });
         std::transform(cc.begin(), cc.end(), std::back_inserter(columns), [](auto& c) { return &c; });
 
-        query::column_id_vector static_columns, regular_columns;
+        query::column_id_vector static_column_ids, regular_column_ids;
 
-        auto sk = column_kind::static_column;
-        auto rk = column_kind::regular_column;
         // TODO: this assumes all mutations touch the same set of columns. This might not be true, and we may need to do more horrible set operation here.
-        for (auto& [r, cids, kind] : { std::tie(p.static_row().get(), static_columns, sk), std::tie(p.clustered_rows().begin()->row().cells(), regular_columns, rk) }) {
-            r.for_each_cell([&](column_id id, const atomic_cell_or_collection&) {
-                auto& cdef =_schema->column_at(kind, id);
-                cids.emplace_back(id);
+        if (!p.static_row().empty()) {
```

Comment:
again - why so intrusive change when it's not needed? Correct me if I'm wrong but you don't seem to change the logic here. If that's true why are you wasting reviewers time? You want to refactor the code without changing the behavior - do this in a separate commit and make it clear what it is. Don't blend it into a complex change.